Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -62,29 +65,35 @@ void LeptonSFProducer::Produce(ZJetEvent const& event,
 {
     LOG(DEBUG) << "\n[" << this->GetProducerId() << "]";
     if (product.m_zValid) {
-        const auto* lep1 = product.m_originalLeptons.at(product.m_zLeptons.first);
-        const auto* lep2 = product.m_originalLeptons.at(product.m_zLeptons.second);
+        const auto* lep1 = m_correctedMuons ? product.m_originalLeptons.at(product.m_zLeptons.first) : product.m_zLeptons.first;
```

Comment:
So I assume there is a new recommendation, to apply these based on fully corrected leptons?

Reply:
Yes that's the recommendation. But in some cases (like turning off Rochester Corrections) one might want to keep the SFs

Reply:
That makes absolutely sense. Keep it like this!